gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -742,16 +742,28 @@ public SeekableByteChannel position(long newPosition) throws IOException {
     if (!isOpen()) {
       throw new ClosedChannelException();
     }
-    checkArgument(newPosition >= 0, "Read position must be non-negative, but was %s", newPosition);
-    checkArgument(
-        newPosition < size(),
-        "Read position must be before end of file (%s), but was %s",
-        size(),
-        newPosition);
+    validatePosition(newPosition, size());
     positionForNextRead = newPosition;
     return this;
   }
 
+  /** Validates that the given position is valid for this channel. */
+  protected void validatePosition(long position, long size) throws IOException {
+    if (position < 0) {
+      throw new EOFException(
+          String.format(
+              "Invalid seek offset: position value (%d) must be >= 0 for '%s'",
+              position, resourceId));
+    }
+
+    if (size >= 0 && position >= size) {
```

Comment:
Can size be less than 0? i.e. isn't the second condition enough?

Reply:
I don't think size should be less than 0, but still better to have it.